sw/device/tests/sim_dv/inject_scramble_seed.c

```diff
@@ -0,0 +1,121 @@
+  CHECK_DIF_OK(dif_otp_ctrl_is_digest_computed(
+      &otp_ctrl, kDifOtpCtrlPartitionSecret1, &secret1_locked));
+
+  if (!secret1_locked && (info & kDifRstmgrResetInfoPor)) {
+    LOG_INFO("Powered up for the first time, program and lock otp");
+    // populate the scramble seeds in otp
+    rand_testutils_reseed();
+
+    // secret partition must be 64b aligned...for some reason.
+    // Figure out that part later
+    for (uint32_t i = 0;
+         i < OTP_CTRL_PARAM_FLASH_ADDR_KEY_SEED_SIZE / WORD_SIZE; ++i) {
+      uint64_t val =
+          (uint64_t)rand_testutils_gen32() << 32 | rand_testutils_gen32();
+      uint32_t addr = (OTP_CTRL_PARAM_FLASH_ADDR_KEY_SEED_OFFSET -
+                       OTP_CTRL_PARAM_SECRET1_OFFSET) +
+                      i * WORD_SIZE;
+      CHECK_DIF_OK(dif_otp_ctrl_dai_program64(
+          &otp_ctrl, kDifOtpCtrlPartitionSecret1, addr, val));
+      otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+    };
+
+    for (uint32_t i = 0;
+         i < OTP_CTRL_PARAM_FLASH_DATA_KEY_SEED_SIZE / WORD_SIZE; ++i) {
+      uint64_t val =
+          (uint64_t)rand_testutils_gen32() << 32 | rand_testutils_gen32();
+      uint32_t addr = (OTP_CTRL_PARAM_FLASH_DATA_KEY_SEED_OFFSET -
+                       OTP_CTRL_PARAM_SECRET1_OFFSET) +
+                      i * WORD_SIZE;
+
+      CHECK_DIF_OK(dif_otp_ctrl_dai_program64(
+          &otp_ctrl, kDifOtpCtrlPartitionSecret1, addr, val));
+      otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+    };
+
+    for (uint32_t i = 0; i < OTP_CTRL_PARAM_SRAM_DATA_KEY_SEED_SIZE / WORD_SIZE;
+         ++i) {
+      uint64_t val =
+          (uint64_t)rand_testutils_gen32() << 32 | rand_testutils_gen32();
+      uint32_t addr = (OTP_CTRL_PARAM_SRAM_DATA_KEY_SEED_OFFSET -
+                       OTP_CTRL_PARAM_SECRET1_OFFSET) +
+                      i * WORD_SIZE;
+      CHECK_DIF_OK(dif_otp_ctrl_dai_program64(
+          &otp_ctrl, kDifOtpCtrlPartitionSecret1, addr, val));
+      otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+    };
+
+    // lock the secret1 partition
+    CHECK_DIF_OK(
+        dif_otp_ctrl_dai_digest(&otp_ctrl, kDifOtpCtrlPartitionSecret1, 0));
+    otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+
+    // inform rom to setup scramble next round
+    uint32_t otp_val = 0;
+    otp_val = bitfield_field32_write(
+        otp_val, FLASH_CTRL_DEFAULT_REGION_SCRAMBLE_EN_FIELD,
+        kMultiBitBool4True);
+    CHECK_DIF_OK(dif_otp_ctrl_dai_program32(
+        &otp_ctrl, kDifOtpCtrlPartitionCreatorSwCfg,
+        (OTP_CTRL_PARAM_CREATOR_SW_CFG_FLASH_DATA_DEFAULT_CFG_OFFSET -
+         OTP_CTRL_PARAM_CREATOR_SW_CFG_OFFSET),
+        otp_val));
+    otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+
+    // reset the device
+    LOG_INFO("Completed first phase, reboot");
+    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+    wait_for_interrupt();
+  } else if (secret1_locked && (info & kDifRstmgrResetInfoPor)) {
+    LOG_INFO("Hello World");
+    return true;
```

Comment:
```suggestion
  } else if (secret1_locked) {
    return true;
```

Reply:
ah i actually wanted the hello world so the DV side can explicitly check this. 
Technically this doesn't add a ton, but i still want the DV side to wait on a check. I'll add a comment to make this clearer.